src/detect-engine-content-inspection.c

```diff
@@ -195,12 +198,14 @@ static int DetectEngineContentInspectionInternal(DetectEngineThreadCtx *det_ctx,
 
                 if (cd->flags & DETECT_CONTENT_DEPTH_VAR) {
                     if ((det_ctx->byte_values[cd->depth] + prev_buffer_offset) < depth) {
+                        distance_included = false;
```

Comment:
Does it make sense to have both distance and depth in a rule ?

Reply:
byte-extract-01 has one rule example with both depth and distance but my understanding is that depth is about the first content being matched, while distance is about the second content being matched

Reply:
And it looks like we have some complex computation when we have a buffer with multiple contents each having its depth/within/distance/etc... to compute for each the global depth inside the buffer